src/Parser/HolyPriest/Modules/PriestCore/Serendipity.js

```diff
@@ -0,0 +1,227 @@
+
+      if (differenceSerenity < this.serendipityReduction * 500) {
+        const overlap = Math.min(
+          this.serendipityReduction * 500,
+          Math.abs((this.serendipityReduction * 500) - differenceSerenity)
+        );
+        this.overcast[SPELLS.HOLY_WORD_SERENITY.id] += overlap;
+        this._tempOvercast[SPELLS.HOLY_WORD_SERENITY.id] += overlap;
+      }
+
+      if (differenceSanctify < this.serendipityReduction * 500) {
+        const overlap = Math.min(
+          this.serendipityReduction * 500,
+          Math.abs((this.serendipityReduction * 500) - differenceSanctify)
+        );
+        this.overcast[SPELLS.HOLY_WORD_SANCTIFY.id] += overlap;
+        this._tempOvercast[SPELLS.HOLY_WORD_SANCTIFY.id] += overlap;
+      }
+
+      // Logic for determining Holy Priest 2P Set Bonus gain
+      if (differenceSerenity > (this.serendipityReduction - SPELLS.HOLY_PRIEST_T20_2SET_BONUS_BUFF.coeff) * 500) {
+        this.holy2PsetValue += Math.min(0.5, differenceSerenity - (this.serendipityReduction - SPELLS.HOLY_PRIEST_T20_2SET_BONUS_BUFF.coeff) * 500);
+      }
+      if (differenceSanctify > (this.serendipityReduction - SPELLS.HOLY_PRIEST_T20_2SET_BONUS_BUFF.coeff) * 500) {
+        this.holy2PsetValue += Math.min(0.5, differenceSanctify - (this.serendipityReduction - SPELLS.HOLY_PRIEST_T20_2SET_BONUS_BUFF.coeff) * 500);
+      }
+
+      this.holyWordCooldowns[SPELLS.HOLY_WORD_SERENITY.id] -= this.serendipityReduction * 500;
+      this.holyWordCooldowns[SPELLS.HOLY_WORD_SANCTIFY.id] -= this.serendipityReduction * 500;
+    }
+  }
+
+  statistic() {
+    const combinedOvercast = (this.overcast[SPELLS.HOLY_WORD_SERENITY.id] + this.overcast[SPELLS.HOLY_WORD_SANCTIFY.id]) / 1000;
+    const formattedOvercastSerenity = formatNumber(this.overcast[SPELLS.HOLY_WORD_SERENITY.id] / 1000);
+    const formattedOvercastSanctify = formatNumber(this.overcast[SPELLS.HOLY_WORD_SANCTIFY.id] / 1000);
+
+    const overcastAsCastsSerenity = (this.overcast[SPELLS.HOLY_WORD_SERENITY.id] / 1000) / this.holyWordMaxCooldown[SPELLS.HOLY_WORD_SERENITY.id];
+    const overcastAsCastsSanctify = (this.overcast[SPELLS.HOLY_WORD_SANCTIFY.id] / 1000) / this.holyWordMaxCooldown[SPELLS.HOLY_WORD_SANCTIFY.id];
+    const overcastAsPercCasts = (
+      (overcastAsCastsSerenity + overcastAsCastsSanctify) / (this.casts[SPELLS.HOLY_WORD_SERENITY.id] + this.casts[SPELLS.HOLY_WORD_SANCTIFY.id] + overcastAsCastsSanctify + overcastAsCastsSerenity)
+    );
+
+    return (
+      <StatisticBox
+        icon={<SpellIcon id={SPELLS.SERENDIPITY.id} />}
+        value={`${formatNumber(combinedOvercast)}s`}
+        label="Wasted Serendipity"
+        tooltip={`${formattedOvercastSerenity} wasted Serenity CD and ${formattedOvercastSanctify} wasted Sanctify CD. This is effectively ${formatPercentage(overcastAsPercCasts)}% of your potential Holy Word casts missed from solely wasted Serendipity.`}
```

Comment:
This still seems like an over-glorified cast efficiency metric to me. Should it really be expressed as 'wasted seconds' rather than just a cast efficiency percent?

Reply:
Would it be better as a percentage of wasted versus effective serendipity? I think that would be good.

Reply:
While it's complicated to calculate the cooldown, gameplay wise does this really play much differently from a a medium static CD? I assume you're casting Flash Heal / PoH because you need the healing, and the Serendipity proc is just a happy side effect. So the Holy Words are ~40 second CD abilities that are a high priority to use when available, typically such a thing would just be baked into the cast efficiency module.

It would make sense to call them 'wasted seconds' if Serendipity was a talent, because you could then suggest the player spec something else if they're not using it. Because it's not a talent, the only action the player can take is have better cast efficiency.

If you'd still prefer to keep this stat, my recommendation would be to change it to "per cast". I also wouldn't mind seeing the average CD reduction in the tooltip or something. Something like "Serendipity Reduction - 17s per Holy Word" ... "Serendipity cut an average of 17s off the CDs of your Holy Word: Sanctify and Holy Word: Serenity" ... "You held your Holy Words for an average of 13s after they became available."

Reply:
```I assume you're casting Flash Heal / PoH because you need the healing, and the Serendipity proc is just a happy side effect.```

This actually isn't always the case. People cast Heal (slow ver.) very commonly to proc Serendipity. People often also cast Flash Heal for Serendipity if it can be brought off CD for heavy ST damage. I can't say it happens too often for Sanctify though, since PoH is quite expensive.

I want to keep this metric more towards some sort of wasted, since anything beneficial is probably not really relevant to know (it doesn't matter how much of a CD was reduced by Serendipity, but it does matter if someone is dumping spells that proc Serendipity while the respective spell is off CD). Since this is also Serendipity related, I imagine something like "average wasted Serendipity" would be a decent alternative (waiting isn't a bad thing since their CDs are decently sized, but as mentioned before, spamming when it's off CD is).

Reply:
I'll defer to your knowledge of the spec. Push the code style changes and I will approve.

Reply:
I've had a lot of extra work at my job as of late, I'll let you know on Discord when I've gotten to doing all of the stylistic/refactor stuff mentioned.